### ISSUE TITLE
predict result is black

### ISSUE THREAD

**Issue by liaojiacai** [milesial/Pytorch-UNet#95]

**Comment by liaojiacai** [comment]
I have a problem, after training finish, the predicted result is black. How can I solve this problem? my predict code as fellow, my test image format is png:
`
def predict_img(net,
                full_img,
                device,
                scale_factor=1,
                out_threshold=0.5,
                use_dense_crf=False):
    net.eval()
    print("type of full_img",type(full_img))
    print("data process")
    
    w, h = full_img.size
    newW, newH = int(scale_factor * w), int(scale_factor * h)
    assert newW > 0 and newH > 0, 'Scale is too small'
    full_img = full_img.resize((newW, newH))

    img_nd = np.array(full_img)

    if len(img_nd.shape) == 2:
        img_nd = np.expand_dims(img_nd, axis=2)

    # HWC to CHW
    img_trans = img_nd.transpose((2, 0, 1))
    if img_trans.max() > 1:
        img_trans = img_trans / 255
    

    img = torch.from_numpy(img_trans)

    img = img.unsqueeze(0)
    img = img.to(device=device, dtype=torch.float32)

    with torch.no_grad():
        output = net(img)

        if net.n_classes > 1:
            probs = F.softmax(output, dim=1)
        else:
            probs = torch.sigmoid(output)

        probs = probs.squeeze(0)

        tf = transforms.Compose(
            [
                transforms.ToPILImage(),
                transforms.Resize(full_img.size[1]),
                transforms.ToTensor()
            ]
        )

        probs = tf(probs.cpu())
        full_mask = probs.squeeze().cpu().numpy()

    if use_dense_crf:
        full_mask = dense_crf(np.array(full_img).astype(np.uint8), full_mask)

    return full_mask > out_threshold
`

**Comment by milesial** [comment]
Have you tried a lower threshold ?
What are the statistics of the probabilities for the output mask before thresholding ?

**Comment by liaojiacai** [comment]
> 
> 
> Have you tried a lower threshold ?
> What are the statistics of the probabilities for the output mask before thresholding ?

thanks, I found the nn.BCEWithLogitsLoss() made it hard to convergence. I solve this problem by replace the loss function.

**Comment by XUYUNYUN666** [comment]
> > Have you tried a lower threshold ?
> > What are the statistics of the probabilities for the output mask before thresholding ?
> 
> thanks, I found the nn.BCEWithLogitsLoss() made it hard to convergence. I solve this problem by replace the loss function.

您好 我也是测出来可视化全是黑色的  我打印出看预测的像素值都为0类  我是二分类0（背景）， 1（血管）， 能加个你的联系方式 请教一下吗 感谢非常感谢

**Comment by leijue222** [comment]
> > Have you tried a lower threshold ?
> > What are the statistics of the probabilities for the output mask before thresholding ?
> 
> thanks, I found the nn.BCEWithLogitsLoss() made it hard to convergence. I solve this problem by replace the loss function.

@liaojiacai Excuse me, which loss function did you use to replace the nn.BCEWithLogitsLoss() ?
